### PR TITLE
Build fix

### PR DESCRIPTION
**Pull request by jburel**
The introduction of the zarr reader leads to a build failure on fresh installation
Problem noticed while preparing the doc
see https://github.com/ome/omero-documentation/actions/runs/2170402809 for example

To test:
* Run locally ``./build.py clean && ./build.py``. You may want to clean the maven cache too.

Edit: [30a7127](https://github.com/ome/openmicroscopy/pull/6312/commits/30a7127cd2ff2b2b88ce73a577ec92ddfbd6c936) builds from source. Check that it is green

### CODE REVIEW

**Comment by jburel** [comment]
[30a7127](https://github.com/ome/openmicroscopy/pull/6312/commits/30a7127cd2ff2b2b88ce73a577ec92ddfbd6c936) builds the source. 
We did not build at all or run the tests (everything is based on the docker images)
This should avoid having a possible broken build without using realising. 
We can decide to review the strategy.

File: .github/workflows/omero_plugin.yml [hunk 1]
```diff
@@ -30,6 +31,30 @@ jobs:
           repository: ome/omero-test-infra
           path: .omero
           ref: ${{ secrets.OMERO_TEST_INFRA_REF }}
+      - uses: actions/setup-python@v2
```
Comment (joshmoore) [inline .github/workflows/omero_plugin.yml]: Would this do what you need as a separate workflow?

**Review by joshmoore — COMMENTED** [review]
Few other comments:
 - I would assume that the Glencoe nexus is mirroring this. Maybe since we sometimes need a mirror _anyway_, it would be an idea to use that as the additional repo? cc: @chris-allan 
 - Do we know why the docker build didn't fail? Is perhaps some flag missing?

File: .github/workflows/omero_plugin.yml (hunk 1 above)
Comment (jburel) [reply .github/workflows/omero_plugin.yml]: I have not tried it to be honest, time was not on my side :-)

**Comment by jburel** [comment]
I need to investigate why it is not working with the docker build itself or why it was commented out. I opted for an option that will give us the confidence

For the dependency, I have followed https://github.com/ome/ZarrReader/blob/main/pom.xml#L44
we could adjust that as required

**Comment by jburel** [comment]
I think I remember now why it has been disabled. When a new method is added for example in the gateway-java, and a test is added to this repository. The version used is a release version of the java-gateway not a specific branch like in CI. It then leads to an error.

File: .github/workflows/omero_plugin.yml (hunk 1 above)
Comment (joshmoore) [reply .github/workflows/omero_plugin.yml]: Ok. Pushed.

**Comment by joshmoore** [comment]
> For the dependency, I have followed https://github.com/ome/ZarrReader/blob/main/pom.xml#L44 we could adjust that as required

Fair enough. Then we can discuss upstream and propagate back here.

File: .github/workflows/omero_plugin.yml (hunk 1 above)
Comment (joshmoore) [reply .github/workflows/omero_plugin.yml]: Interestingly this is now failing with:

```
copy-client:
Created dir: /home/runner/work/openmicroscopy/openmicroscopy/dist/lib/client
:: problems summary ::
:::: WARNINGS
		[NOT FOUND  ] io.netty#netty-transport-native-epoll;4.1.46.Final!netty-transport-native-epoll.jar (0ms)
	==== user-maven: tried
	  /home/runner/.m2/repository/io/netty/netty-transport-native-epoll/4.1.46.Final/netty-transport-native-epoll-4.1.46.Final.jar
		[NOT FOUND  ] org.kohsuke#file-leak-detector;1.13!file-leak-detector.jar (0ms)
	==== user-maven: tried
	  /home/runner/.m2/repository/org/kohsuke/file-leak-detector/1.13/file-leak-detector-1.13.jar
		::::::::::::::::::::::::::::::::::::::::::::::
		::              FAILED DOWNLOADS            ::
		:: ^ see resolution messages for details  ^ ::
		::::::::::::::::::::::::::::::::::::::::::::::
		:: io.netty#netty-transport-native-epoll;4.1.46.Final!netty-transport-native-epoll.jar
		:: org.kohsuke#file-leak-detector;1.13!file-leak-detector.jar
		::::::::::::::::::::::::::::::::::::::::::::::
```

Comment (sbesson) [reply .github/workflows/omero_plugin.yml]: Interesting, `file-leak-detector` is a dependency from the `test-suite` component of ome/bioformats for detecting unclosed file handles.

I suspect it is being pulled as a result of https://github.com/ome/ZarrReader/blob/20ccbe52ff68bb8f7af0ed460c6cc29b0e31ffb8/pom.xml#L124-L127. Assuming nothing depends on it (which arguably would be a bug), either this dependency should be removed completely from the POM or it should be declared within the `test` scope`.

**Comment by jburel** [comment]
@joshmoore I am working on fixes to re-activate the docker build and hopefully get something working. Obviously the error above is a problem. Strange that it did not happen before

**Comment by jburel** [comment]
Interesting that one build and the other one does not

File: .github/workflows/omero_plugin.yml (hunk 1 above)
Comment (jburel) [reply .github/workflows/omero_plugin.yml]: Adding ``<scope>test</scope>`` leads to an error when running ``mvn install``
```
ZarrReader.java:[640,15] cannot find symbol
  symbol:   variable StringUtils
```
the apache dependency is then missing.

**Comment by jburel** [comment]
One more to go
```
		::::::::::::::::::::::::::::::::::::::::::::::
		::              FAILED DOWNLOADS            ::
		:: ^ see resolution messages for details  ^ ::
		::::::::::::::::::::::::::::::::::::::::::::::
		:: io.netty#netty-transport-native-epoll;4.1.46.Final!netty-transport-native-epoll.jar
```
Coming from https://github.com/ome/ZarrReader/blob/main/pom.xml#L85
The version used has numerous CVEs

**Comment by jburel** [comment]
Replaced by https://github.com/ome/openmicroscopy/pull/6313

**Pull request closed without merging**